### DISCUSSION TITLE
Allow Url Providers to include query string components and not conflict with base URL

### DISCUSSION THREAD

**Discussion by NikRimington** [umbraco/Umbraco-CMS#16620] [Features and ideas]
Currently, due to how the collision detection works, it is not possible to have a URL Provider generate a query string derived URL.

For example the following tree structure generates the associated URLs:

- Home -- /
  - Listing Page -- /listing-page
    - Sub Cat A -- /listing-page/sub-cat-a
    - Sub Cat B -- /listing-page/sub-cat-b

What I attempted to do was add a URL Provider only for the "Sub Category" page type that changed their URL to the following pattern:

`{parent-url}?filter={node name}` e.g `/listing-page?filter=sub cat a`

However, as soon as a URL provider that did this was registered the collision detection code in the Default URL Provider triggered and in the back office it reported that the URL collided with another node (the parent node).

Ideally, I would like to see an option added that allowed URL's from providers to be flagged as to if they should or shouldn't perform the URL collision detection to avoid this scenario.

**Comment by james-whittington1** [comment]
Another issue here is that when you pick this document in the link picker (eg, a link picker on the home page pointing at a sub cat a), it will add the query string to the "query string' section of the picker. When you try to use the link pickers value in your code, the URL provider is used to get the URL (as expected, it will contain the query string), and then the query string from the link picker is also appended. 

The only way I found was to check if the current Umbraco request is a front end request in my url provider which seems a bit hack but gives a satisfactory result for my use case.
